### PR TITLE
Handle aliased imports in `FullyQualifiedImport`

### PR DESCRIPTION
**Pull request by Cirras**
This PR fixes an issue in `FullyQualifiedImport` where unit aliases weren't considered, causing false positives on aliased imports.

### CODE REVIEW

File: delphi-checks/src/test/java/au/com/integradev/delphi/checks/FullyQualifiedImportCheckTest.java [hunk 1]
```diff
@@ -82,6 +82,18 @@ void testUnresolvedImportShouldNotAddIssue() {
...
   }
 
+  @Test
+  void testUnitAliasImportShouldNotAddIssue() {
+    var testFile = new DelphiTestUnitBuilder().appendDecl("uses").appendDecl("  AliasName;");
+
+    CheckVerifier.newVerifier()
+        .withCheck(new FullyQualifiedImportCheck())
+        .withSearchPathUnit(new DelphiTestUnitBuilder().unitName("Scope.RealName"))
+        .withUnitAlias("AliasName", "Scope.RealName")
```
Comment (zaneduffield) [inline delphi-checks/src/test/java/au/com/integradev/delphi/checks/FullyQualifiedImportCheckTest.java:92]: I think this test case could be a bit stronger. The problem I see is that the test will continue to pass if the unit alias isn't working at all; the import will be unresolved and no issue will be raised.

You could *also* add the unit scope name of `Scope` alongside a new unit `Scope.AliasName`, which will cause an issue to be raised if the alias is not working (but the Check *is*).

```suggestion
        .withSearchPathUnit(new DelphiTestUnitBuilder().unitName("Scope.RealName"))
        .withSearchPathUnit(new DelphiTestUnitBuilder().unitName("Scope.AliasName"))
        .withUnitScopeName("Scope")
        .withUnitAlias("AliasName", "Scope.RealName")
```

File: delphi-checks/src/main/java/au/com/integradev/delphi/checks/FullyQualifiedImportCheck.java [hunk 2]
```diff
@@ -25,35 +25,40 @@
...
+                  .withEdit(QuickFixEdit.replace(unitImportNode.getNameNode(), expected)))
           .report();
     }
 
     return context;
   }
+
+  private static boolean isAliasImport(
+      UnitImportNameDeclaration importDeclaration, UnitNameDeclaration unitDeclaration) {
+    return !importDeclaration.simpleName().equalsIgnoreCase(unitDeclaration.simpleName());
```
Comment (zaneduffield) [inline delphi-checks/src/main/java/au/com/integradev/delphi/checks/FullyQualifiedImportCheck.java:62]: What if you alias `Scope.Name` to `Name`?

**Review by zaneduffield — CHANGES_REQUESTED** [review]
One extra edge case to consider, and a possible improvement to the test.

File: delphi-checks/src/main/java/au/com/integradev/delphi/checks/FullyQualifiedImportCheck.java (hunk 2 above)
Comment (Cirras) [reply delphi-checks/src/main/java/au/com/integradev/delphi/checks/FullyQualifiedImportCheck.java:62]: Uh oh, this is a bit of an annoying headache to detect.
Guess it's time for a new API method, like `UnitImportNameDeclaration::isAliased`.

**Review by zaneduffield — APPROVED** [review]
Looks good! 👍
